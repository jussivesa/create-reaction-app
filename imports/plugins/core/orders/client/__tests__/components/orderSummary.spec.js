jest.mock("/imports/plugins/core/ui/client/components", () => ({
  Badge() { return null; },
  ClickToCopy() { return null; }
}));

import React from "react";
import { shallow } from "enzyme";
import shallowToJSON from "enzyme-to-json";
import OrderSummary from "../../components/orderSummary";

/**
 * Order Summary is a display only component
 * It receives props and displays them accordingly
 */

afterEach(() => {
  jest.clearAllMocks();
});

/**
 * Snapshots make sure your UI does not change unexpectedly
 */

test("OrderSummary snapshot test", () => {
  // Initializing all the props passed into order summary component
  const dateFormat = jest.fn();
  const tracking = jest.fn();
  const shipmentStatus = jest.fn(() => ({}));
  const printableLabels = jest.fn(() => ({}));
  const profileShippingAddress = {};
  const order = {
    shipping: [{ shipmentMethod: {} }],
    workflow: {
      status: "new"
    },
    billing: [
      {
        paymentMethod: {},
        invoice: {}
      }
    ]
  };

  const component = shallow((
    <OrderSummary
      dateFormat={dateFormat}
      tracking={tracking}
      profileShippingAddress={profileShippingAddress}
      shipmentStatus={shipmentStatus}
      printableLabels={printableLabels}
      order={order}
    />
  ));
  const tree = shallowToJSON(component);
  expect(tree).toMatchSnapshot();
});
