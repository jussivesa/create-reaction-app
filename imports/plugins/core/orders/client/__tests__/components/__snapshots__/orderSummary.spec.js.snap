// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary snapshot test 1`] = `
<div>
  <div
    className="order-summary-form-group bg-info"
    style={
      Object {
        "lineHeight": 3,
        "marginLeft": -15,
        "marginRight": -15,
        "marginTop": -15,
      }
    }
  >
    <strong
      style={
        Object {
          "marginLeft": 15,
        }
      }
    />
    <div
      className="invoice-details"
      style={
        Object {
          "marginRight": 15,
          "position": "relative",
        }
      }
    />
  </div>
  <div
    className="roll-up-invoice-list"
  >
    <div
      className="roll-up-content"
    >
      <div
        style={
          Object {
            "marginBottom": 4,
          }
        }
      >
        <Badge
          badgeSize="large"
          i18nKeyLabel="cartDrawer.new"
          label="new"
          status="info"
        />
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="order.orderId"
        >
          Order ID
        </strong>
        <div
          className="invoice-details"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <ClickToCopy
            i18nKeyTooltip="admin.orderWorkflow.summary.copyOrderLink"
            tooltip="Copy Order Link"
          />
        </div>
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="order.created"
        >
          Created
        </strong>
        <div
          className="invoice-details"
        >
          a few seconds ago
           | 
        </div>
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="order.processor"
        >
          Processor
        </strong>
        <div
          className="invoice-details"
        />
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="order.payment"
        >
          Payment
        </strong>
        <div
          className="invoice-details"
        >
           (
          )
        </div>
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="order.transaction"
        >
          Transaction
        </strong>
        <div
          className="invoice-details"
        />
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="orderShipping.carrier"
        >
          Carrier
        </strong>
        <div
          className="invoice-details"
        >
           - 
        </div>
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="orderShipping.tracking"
        >
          Tracking
        </strong>
        <div
          className="invoice-details"
        />
      </div>
      <div
        className="order-summary-form-group"
      >
        <strong
          data-i18n="orderShipping.printLabels"
        >
          Labels
        </strong>
        <a
          className="invoice-details"
          target="_blank"
        >
          <span
            data-i18n="orderShipping.printCustomsLabel"
          >
            Print Customs
          </span>
        </a>
      </div>
    </div>
  </div>
  <br />
  <div
    className="order-summary-form-group"
  >
    <strong
      data-i18n="orderShipping.shipTo"
    >
      Ship to
    </strong>
    <div
      className="invoice-details"
    >
      <strong>
        Phone: 
      </strong>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": 4,
      }
    }
  >
    <span />
    <br />
    <span />
    <br />
    <span>
      , 
      , 
       
    </span>
  </div>
</div>
`;
